### DISCUSSION TITLE
question about fetch depth

### DISCUSSION THREAD

**Discussion by DxWang36** [microsoft/PR-Metrics#524] [Q&A]
when I run the PR-Metrics Task, it report a error
##[error]Could not access sufficient Git history. Set 'fetchDepth: 0' as a parameter to the 'checkout' task (YAML) or disable 'Shallow fetch' under the build process phase settings (classic).
So it means that I must use fetchDepth:0 ?  my code repo is quite big, can I just pull a little commit?

**Comment by DxWang36** [comment]
My code repo is quite big, like 80G size, so can i set the fetchDepth = 1 or fetchDepth=200 ?

**Comment by muiriswoulfe** [comment]
Hi @DxWang36,

The fetch depth constraint comes from Git. To execute the command for determining the set of changes, Git requires certain historical Git commit data. It's hard to give a definitive answer as to how much history is required due to the way Git works, and due to differences in the way commits could be made.

If `fetchDepth: 0` is a problem, my advice would be to try different fetch depths. The error reported will always be the same as `fetchDepth: 0` is the only value guaranteed to work. But if the Git command succeeds, you won't see that error message.

If that doesn't work, the only solution would be to flatten some of the Git commit history, but this probably isn't advisable as it would change the commit info.

Hope this helps.

**Reply by DxWang36** [reply]
Hi Woulfe,

Thanks for your help, I will try some different depth, Maybe this tool is not fit for our repo(too big), we can not use fetchDepth:0

Thanks
